File: lily/basic-types/src/main/java/net/bodz/lily/schema/geo/Zone.java
package net.bodz.lily.schema.geo;

import javax.persistence.Table;

import net.bodz.bas.meta.res.HaveAttachments;
import net.bodz.lily.concrete.util.GeoLocation;
import net.bodz.lily.entity.attachment.util.IImagesInProps;

@HaveAttachments
@Table(schema = Zone.SCHEMA_NAME, name = Zone.TABLE_NAME)
public class Zone
        extends _Zone_stuff
        implements
            IImagesInProps {

    private static final long serialVersionUID = 1L;

    public static final int ID_World = 0;
    public static final int ID_China = 86;

    public final GeoLocation geo = new GeoLocation();
    private String fullPath = "";

    public GeoLocation getGeo() {
        return geo;
    }

    public String getFullPath() {
        return fullPath;
    }

    public void setFullPath(String fullPath) {
        this.fullPath = fullPath;
    }

}

File: lily/@lily/packages/basic/src/net/bodz/lily/schema/geo/Zone.ts
import GeoLocation from "../../concrete/util/GeoLocation";
import ZoneTypeInfo from "./ZoneTypeInfo";
import _Zone_stuff from "./_Zone_stuff";

export class Zone extends _Zone_stuff {

    static _typeInfo: ZoneTypeInfo;
    static get TYPE() {
        if (this._typeInfo == null)
            this._typeInfo = ZoneTypeInfo.INSTANCE;
        return this._typeInfo;
    }

    fullPath?: string
    geo?: GeoLocation

    constructor(o?: any) {
        super(o);
    }
}

export default Zone;
